Improve code context menu layout position esp visual stability

mgsloan opened the pull request:
* Now decides whether the menu is above or below the target position before rendering it. This causes its position to no longer vary depending on the length of completions

* When the text area is height constrained (< 12) lines, now chooses the side which has the most space. Before it would always display above if height constrained below.

* Misc code cleanups

Release Notes:

- Improved completions menu layout to be more stable and use available space better.

maxdeviant commented on crates/editor/src/code_context_menus.rs:
```diff
@@ -859,8 +891,10 @@ impl CodeActionsMenu {
         .elevation_1(cx)
-        .p_1()
-        .max_h(max_height)
+        // x padding is the same as y.
```
We could use a single `.p` here:

```suggestion
        .p(Self::y_padding() / 2.)
```

mgsloan replied:
Ah true, thanks!!  Ended up replacing this code with use of `ListItem` + `Popover` in #22112

mgsloan committed:
Improve code context menu layout position esp visual stability

* Now decides whether the menu is above or below the target position
before rendering it. This causes its position to no longer vary
depending on the length of completions

* When the text area is height constrained (< 12) lines, now chooses
the side which has the most space. Before it would always display
above if height constrained below.

* Misc code cleanups
